### PR TITLE
Allow different base activations per location in checker and interpreter

### PR DESCRIPTION
**Pull request by turbolent**
## Description

Allow types and values of an environment to be declared in specific locations.

Change the checker and interpreters configurations from values to handlers, which can return different base activations for different locations.

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by SupunS — APPROVED** [review]

File: runtime/environment.go [hunk 1]
```diff
@@ -1096,3 +1140,31 @@ func (e *interpreterEnvironment) CommitStorage(inter *interpreter.Interpreter) e
 
 	return nil
 }
+
+func (e *interpreterEnvironment) getBaseValueActivation(location common.Location) (activation *sema.VariableActivation) {
```
[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/environment.go:1144]: The idea here is to have a base activation depending on the location. A host might want a custom base activation with specific built-in / pre-declared values / types, for a specific location. 

`activations[nil]` is the base activation for `location == nil`, i.e. the one that is used by default for all locations for which no custom base activation was declared (the custom base activation is implicitly created by the environment, when the host declares a value / type for a specific location, see the functions above).

I'll rename and document the `activations` variables and fields to make it clear they are not e.g. activation stacks, all values in the maps are base activations, and the maps / base activations are "per location".

[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/environment.go]: @SupunS Improved this n 135419123ede775caf9d1e95fe033d2dae11292a and e44edd777c907def2804b072432a2d54530d33ab. Can you please have another look?

[the remark this answers is not in the document]
Comment (SupunS) [reply runtime/environment.go]: ah ok, that makes sense. Thanks for the explanation and the refactor 👌

**Review by SupunS — APPROVED** [review]

**Comment by turbolent** [comment]
@dsainati1 Could you please also have a look? This unlocks implementation of the Flex/EVM contract

**Review by dsainati1 — APPROVED** [review]

**Pull request merged**